Repository searching examples (#1361)
PowerShell/PSResourceGet: Examples/FindRepositorySearchingExamples.md

```

1) Searching with only a package name specified, eg: `Find-PSResource 'TestModule'` or `Find-PSResource 'TestModule' -Repository '*'`
    * When the package exists in both repositories:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        TestModule  1.0.0.0            NuGetGallery 
        ```
       Should return 'TestModule' from both 'PSGallery' and 'NuGetGallery'.
       
    * When the package is in the first repository, but not the second:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        ```
        Should return 'TestModule' from 'PSGallery'.

    * When the package is the second repository, but not the first:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            NuGetGallery 
        ```
        Should return 'TestModule' from 'NuGetGallery'.

    * When the package is in neither repository:
        ```

>         Find-PSResource: Package 'TestModule' could not be found.

Reviewer: ```suggestion
        Find-PSResource: Package 'TestModule' could not be found in any registered repositories.
```

        ```
2) Searching with a package name and a repository specified, eg: `Find-PSResource 'TestModule' -Repository PSGallery`
    * When the package exists in both repositories:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        ```
        Should return 'TestModule' from 'PSGallery'.

>     * When the package is in the first repository, but not the second:

Reviewer: This feels a bit confusing, because whether the package exists in NuGetGallery or not we would not search it. For scenario number 2 you could either a) have 2 examples (package exists in PSGallery, package does not exist in PSGallery). Because here your 2nd and 3rd example are the same, so you can just highlight this simpler relationship/aspect. Or b) if you stick with the current format use parenthesis to indicate which repo is what for "first" and "second" because I had to look back and remind myself whether first or second was PSGallery.

```suggestion
    * When the package is in the first repository (PSGallery), but not the second (NuGetGallery):
```

        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        ```
        Should return 'TestModule' from 'PSGallery'.
        
    * When the package is the second repository, but not the first:
        ```

>         Find-PSResource: Package 'TestModule' could not be found.

Reviewer: ```suggestion
        Package with name 'TestModule' could not be found in repository 'PSGallery'.
```

        ```
    * When the package is in neither repository:
        ```

>         Find-PSResource: Package 'TestModule' could not be found.

Reviewer: ```suggestion
        Package with name 'TestModule' could not be found in repository 'PSGallery'.
```

Reviewer: this is the correct error message.

        ```
        
3) Searching with a package name specified and wildcard repository, eg: `Find-PSResource 'TestModule' -Repository *Gallery`
    * When the package exists in both repositories:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        TestModule  1.0.0.0            NuGetGallery 
        ```
        Should return 'TestModule' from 'PSGallery' and 'NuGetGallery'.
        

>     * When the package is in the first repository, but not the second:

Reviewer: ```suggestion
    * When the package exists in the first repository, but not the second:
```

        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        ```
        Should return 'TestModule' from 'PSGallery'.
        

>     * When the package is the second repository, but not the first:

Reviewer: ```suggestion
    * When the package exists in the second repository, but not the first:
```

        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            NuGetGallery 
        ```
        Should return 'TestModule' from 'NuGetGallery'.
        
    * When the package is in neither repository:
        ```

>         Find-PSResource: Package 'TestModule' could not be found.

Reviewer: ```suggestion
        Package 'TestModule' could not be found in registered repositories: 'PSGallery, NuGetGallery'.
```

Reviewer: this is the correct error message.

        ```
        
4) Searching with a package name specified and multiple repository names specified, eg: `Find-PSResource 'TestModule' -Repository PSGallery, NuGetGallery`

    * When the package exists in both repositories:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        TestModule  1.0.0.0            NuGetGallery 
        ```
        Should return 'TestModule' from 'PSGallery' and 'NuGetGallery'.
        
    * When the package is in the first repository, but not the second:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        ```

>         Should return 'TestModule' from 'PSGallery'.

Reviewer: ```suggestion
        Should return 'TestModule' from 'PSGallery'.
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'NuGetGallery'.
```

        
    * When the package is the second repository, but not the first:
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            NuGetGallery 
        ```

>         Should return 'TestModule' from 'NuGetGallery'.

Reviewer: ```suggestion
        Should return 'TestModule' from 'NuGetGallery'.
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'PSGallery'.
```

        
    * When the package is in neither repository:
        ```

>         Find-PSResource: Package 'TestModule' could not be found.

Reviewer: ```suggestion
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'PSGallery'.
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'NuGetGallery'.
```

        ```
        
5) Searching with a package name specified and both a repository name specified AND a repository name with a wildcard, eg: `Find-PSResource 'TestModule' -Repository *Gallery, LocalRepo`

>     * For all scenarios:

Reviewer: what is this scenario?

Author: Do you mean what is “all scenarios”? It means the package could be in any or all repository and the results would still be the same, the error below.

Author: this is updated now!

Author: (as per our conversation offline)

        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            PSGallery 
        
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'NuGetGallery'.
        ```
        Should return 'TestModule' from 'PSGallery'.
       
        
    * When the package exists the second repository (NuGetGallery), but not the first (PSGallery):
        ```
        Name        Version Prerelease Repository
        ----        ------- ---------- ----------
        TestModule  1.0.0.0            NuGetGallery 
        
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'PSGallery'.
        ```
        Should return 'TestModule' from 'NuGetGallery'.
        
        
    * When the package is in neither repository:
        ```
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'PSGallery'.
        Find-PSResource: Package with name 'TestModule' could not be found in repository 'NuGetGallery'.
        ```
        

> 5) Searching with a package name specified and both a repository name specified AND a repository name with a wildcard, eg: `Find-PSResource 'TestModule' -Repository *Gallery, PSGallery`

Reviewer: ```suggestion
5) Searching with a package name specified and both a repository name specified AND a repository name with a wildcard, eg: `Find-PSResource 'TestModule' -Repository *Gallery, otherRepository`
```
